src/containers/Tenant/QueryEditor/QueryEditor.js

```diff
@@ -150,21 +152,37 @@ function QueryEditor(props) {
     }, [props.executeQuery, props.executeQuery]);
 
     useEffect(() => {
-        const {
-            path,
-            executeQuery: {input},
-        } = props;
-        if (resultType === RESULT_TYPES.EXPLAIN) {
-            props.getExplainQuery({query: input, database: path});
+        const {monacoHotKey, setMonacoHotKey} = props;
+        setMonacoHotKey(null);
```

Comment:
No, it won't be endless loop: 1) if useEffect receives monacoHotKey prop not null, it set it to null (156 line) and gets to switch-case statement.
2) After that useEffect will be invoked again with monacoHotKey === null and sets in to null again (156 line).
3) And next time useEffect will not be invoked because null === null.

But I see that step 2 is redundant. It will be more correct to just return if monacoHotKey is null.